Plugins~/Src/MeshSyncClientBlender/python/unity_mesh_sync_preferences.py

```diff
@@ -18,6 +20,73 @@ def msb_get_editor_path_prefix_default():
         path = "/Applications/Unity/Hub/Editor"
     return path
 
+class MESHSYNC_OT_OpenHub(bpy.types.Operator):
+    bl_idname = "meshsync.open_hub"
+    bl_label = "Select Project with Unity Hub"
+    active = False
+    logs_thread = None
+
+
+    @classmethod
+    def description(cls, context, properties):
+        return "Open Unity Hub to select a Unity project to sync data to"
+
+    def invoke(self, context, event):
+
+        if MESHSYNC_OT_OpenHub.active:
+            MESHSYNC_OT_OpenHub.active = False
+            if self.logs_thread is not None:
```

Comment:
Might be worth reading the python docs about using python threads to ensure you get the behavior you need: https://docs.python.org/3/library/threading.html. From reading it seems to suggest these logs_threads should possible be marked as daemons, and futhermore joining without a minimum timeout might cause this code to hang indefinitely, which could happen in the scenario where log_file.readlines() unexpectedly hangs. A pattern I've seen work is a generous timeout period then an abort where the thread has failed to stop in that time. Might be overly cautious, depending on whether or not this code hanging could hang the blender main thread.

Reply:
I made some improvements based on the points you raised. 
1. The threads are now marked as daemons.
2. added a timeout of 1s for joining threads
3. replaced readlines with a readline loop. I check the exit condition in the loop to avoid long hangs with readlines.
4. I registered a shutdown function to the atexit event. The function will attempt to join the thread(with a 1s timeout) when the program exits.